backend/projects/views/dimension.py

```diff
@@ -0,0 +1,59 @@
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
+
+from projects.models import DynamicDimension, DimensionMetaData, ProjectDimension
+from projects.permissions import IsProjectAdmin, IsProjectStaffAndReadOnly
+from projects.serializers import DynamicDimensionSerializer, DimensionMetaDataSerializer, ProjectDimensionSerializer
+from label_types.models import ScaleType
+
+
+class DimensionMetaDataList(generics.ListCreateAPIView):
+    serializer_class = DimensionMetaDataSerializer
+    pagination_class = None
+    permission_classes = [IsAuthenticated & (IsProjectAdmin | IsProjectStaffAndReadOnly)]
+
+    def get_queryset(self):
+        return DimensionMetaData.objects.filter(dimension=self.kwargs["dimension_id"])
+
+
+class DynamicDimensionList(generics.ListCreateAPIView):
+    serializer_class = DynamicDimensionSerializer
+    pagination_class = None
+    permission_classes = [IsAuthenticated & (IsProjectAdmin | IsProjectStaffAndReadOnly)]
+
+    def get_queryset(self):
+        return DynamicDimension.objects.all()
+
+    def perform_create(self, serializer):
+        serializer.save()
+
```

Comment:
I'm sorry, I just noticed a bug when testing with the FE together with the BE. When I create a new project and assign the dimensions later (basically using the view `AddProjectDimension`, or in other words the `/assign_dimensions` endpoint), the scale label is properly created in that new project.

However, when I create a new project and assign the dimensions in the project creation process (like in the screenshot below -- notice that it is on the project creation page), the scale label for the "testDIM" is not created. Because of this, user can do annotation just fine on the annotation page, but the annotation for that particular dimension won't get saved into DB.
<img width="960" alt="Untitled" src="https://user-images.githubusercontent.com/64476430/227923425-2bbd388f-1779-4a74-b866-c9c1c5c2ff72.png">

Therefore, I think the `perform_create()` method of `DynamicDimensionList` also requires a similar solution of
```
if dimension.id > 69 and dimension.type=='slider':
  ScaleType.objects.create(text=dimension.name, project_id=project_id)
```

Would it be possible?

Reply:
Yeah, thank you for finding it. It's my mistake when I added the ability to assign newly created dimension to project creation step, I forgot to update the ability to assign the scale label automatically :sweat_smile: I updated it, result is as below:
Project dimension table:
![Screenshot from 2023-03-27 13-39-19](https://user-images.githubusercontent.com/17596118/227931774-f74b1dce-1863-41ca-a6d2-9804c11a9c67.png)

Scale table:
![Screenshot from 2023-03-27 13-39-37](https://user-images.githubusercontent.com/17596118/227931864-537c0110-193d-45f4-8ffa-9a71e4c20d18.png)